### PR TITLE
Callback queue

### PR DESCRIPTION
**Pull request by trapajim**
i've added a simple fifo queue for the callbacks. 
requested in #19

### CODE REVIEW

File: examples/callbacks/callbacks.go [hunk 1]
```diff
@@ -4,7 +4,7 @@ import (
 	"fmt"
 	"time"
 
-	"github.com/muesli/cache2go"
+	"github.com/trapajim/cache2go"
```
Comment (muesli) [inline examples/callbacks/callbacks.go]: Careful, I assume you accidentally committed that change. Please revert it.

**Comment by muesli** [comment]
Nice contribution, thank you!

File: examples/callbacks/callbacks.go [hunk 2]
```diff
@@ -35,6 +38,8 @@ func main() {
 	// Deleting the item will execute the AboutToDeleteItem callback.
 	cache.Delete("someKey")
 
+	//remove AddedItemCallbacks
```
Comment (muesli) [inline examples/callbacks/callbacks.go]: That comment feels a bit redundant :smile:

File: examples/callbacks/callbacks.go (hunk 1 above)
Comment (trapajim) [reply examples/callbacks/callbacks.go]: sorry for that. I've changed it.

File: examples/callbacks/callbacks.go (hunk 2 above)
Comment (trapajim) [reply examples/callbacks/callbacks.go]: not sure what i've thought there, I removed the comment.

**Comment by muesli** [comment]
Awesome, thanks for updating. One more thing occurred to me in the meantime though:

this could lead to breaking existing clients, if they relied on the behavior of the old callbacks being overwritten.

Here's an idea: how about we keep the `set*Callback` methods overwriting the entire list of callbacks, and add new `add*Callback` methods for the new functionality?

**Comment by trapajim** [comment]
> Awesome, thanks for updating. One more thing occurred to me in the meantime though:
> 
> this could lead to breaking existing clients, if they relied on the behavior of the old callbacks being overwritten.
> 
> Here's an idea: how about we keep the `set*Callback` methods overwriting the entire list of callbacks, and add new `add*Callback` methods for the new functionality?

That makes sense, I didn't think about that. I'll implement that soon.

**Comment by trapajim** [comment]
@muesli just checking in if you saw the changes

**Comment by muesli** [comment]
@trapajim I'm terribly sorry, it seems I have really missed that notification! Thanks for your contributions, really appreciated!

**Pull request merged**